Hot-Rod-Transactions-NonXA.asciidoc

```diff
@@ -0,0 +1,311 @@
+|Lifespan
+|vInt
+|Number of seconds that a entry during which the entry is allowed to life.
+If number of seconds is bigger than 30 days, this number of seconds is treated as UNIX time and so, represents the number of seconds since 1/1/1970.
+If set to `0`, lifespan is unlimited.
+
+|Max Idle
+|vInt
+|Number of seconds that a entry can be idle before it’s evicted from the cache.
+If `0`, no max idle time.
+
+|Value Length
+|vInt
+|Length of value
+
+|Value
+|byte-array
+|Value to be stored
+
+|===
+
+.Response
+[cols="<.^,<.^,<"]
+|===
+|Field Name |Size |Value
+
+|Header
+|variable
+|Response header
+
+|Status
+|byte
+|One of the following values +
++0x00+ OK +
++0x01+ Invalid configuration (server side) +
+
+|XA return code
+|vInt
+|The XA code representing the prepare response. +
+If `Status==0x00`, this field value is
+`XA_OK(0)`, `XA_RDONLY(3)` or any of the error codes (see `XaException`) +
+If `Status==0x01`, this field is ignored.
+
+|===
+
+=== `COMMIT_TX` and `COMMIT_TX_RESPONSE`
+
+.Request
+[cols="<.^,<.^,<"]
+|===
+|Field Name |Size |Value
+
+|Header
+|variable
+|Request header
+
+|===
+
+The response is similar and it will probably reuse the same response for all the requests.
+
+.Response
+[cols="<.^,<.^,<"]
+|===
+|Field Name |Size |Value
+
+|Header
+|variable
+|Response header
+
+|Status
+|byte
+|One of the following values +
++0x00+: OK +
++0x01+: Invalid configuration (server side) +
+
+|XA return code
+|vInt
+|The XA code representing the prepare response. +
+If `Status==0x00`, this field value is
+`XA_OK(0)` or any of the error codes (see `XaException`) +
+If `Status==0x01`, this field is ignored.
+|===
+
+The `ROLLBACK_TX` and `ROLLBACK_TX_RESPONSE` operations has the same structure.
+
+== Failure Scenarios
+
+Nodes crashing or topology changes in the server doesn't affect the transaction during its execution.
+However, some scenarios during the transaction prepare phase need to be handled.
+Note that, after the transaction is prepared, the commit or rollback phase can happen in any node and it is free to be retried.
+
+=== Topology change during transaction replay and prepare phase
+
+Nodes joining or leaving during the transaction replay is already handled.
+However, a special attention is needed if the originator crashes.
+
+The client is allowed to retry the prepare phase in another node, but the server needs to ensure that the `GlobalTransaction` is the same.
+This requirement is needed for `PESSIMISTIC` locking mode due to the lock acquisition at transaction runtime.
+
+This solution covers the case where the client sends the prepare and the server crashes without sending the reply.
+
+=== Topology change during transaction second phase
+
+In this scenario, the client has the outcome of the prepare phase and it is free to retry the commit or rollback request in whatever node it wants.
+
+=== Client crashes/disconnects
+
+After a prepare phase, the locks are held until the client sends the commit or rollback request.
```

Comment:
But I don't think that the transaction manager's timeout creates inconsistencies. If the transaction takes too long to run, it is safe to rollback. 
In our case, the transaction is prepared and it is bad the transaction manager requires the commit and we reply that "the transaction was aborted, sorry"